### PR TITLE
[openstack_manila] add new plugin at sos/plugins

### PR DESCRIPTION
**Pull request by poornimakshirsagar**
Capture configuration and log files for OpenStack Manila project
https://wiki.openstack.org/wiki/Manila

Manila is File Share Service prototype, which provides coordinated
access to shared or distributed file systems. This plugin captures
manila related logs from openstack system.

Signed-off-by: Poornima M. Kshirsagar pkshiras@redhat.com

### CODE REVIEW

File: sos/plugins/openstack_manila.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    packages = (
+        'python-manila',
+        'manila-common',
+        'manila-api',
+        'manila-share',
+        'manila-scheduler'
+    )
+
+    def setup(self):
+        super(DebianManila, self).setup()
```
Comment (bmr-cymru) [inline sos/plugins/openstack_manila.py]: No need to define a `setup()` method if the only thing it does is to call the superclass' `setup()` - if you just omit the method from the derived class Python will automatically bind the name to the one supplied by the parent.

Comment (poornimakshirsagar) [inline sos/plugins/openstack_manila.py]: Thanks for review @bmr-cymru  will omit setup() method here.

**Comment by ervikrant06** [comment]
@poornimakshirsagar I have written this plugin few months back. Already PL for that in pipeline. You may have missed my PL.

https://github.com/sosreport/sos/pull/864

**Comment by poornimakshirsagar** [comment]
Hi @ervikrant06. It seems you have forget to ensure password protect_keys in manila.conf over #864 
I have dev-tested #906  which captures desired logs .  Let me know if you want any changes around this plugin.

**Comment by ervikrant06** [comment]
@poornimakshirsagar thanks for pointing about protect_keys.  Isn't we are supposed to modify the oldest one? You may have made the change in it.

**Comment by bmr-cymru** [comment]
As @poornimakshirsagar already has the password processing implemented I'm inclined to take this for master - that said, there are a couple of things that could do with adding from the version that @ervikrant06 already submitted (and the associated review):

* Place the plugin in the `storage` profile
* `add_cmd_output("manilla list")`
* Check for `OS_*` environment variables (I assume these are used for `manilla list`?)

If these are useful then it would be good to get them into the final plugin - I am fine with follow-ups to this PR (as it has the password bits in it it is ready to merge now), or amending and extending the existing branch.

I'll leave this open a few days for you to decide.

**Comment by poornimakshirsagar** [comment]
Well @ervikrant06 I was not aware of your old PR , did searched for any existing Manila stuff there in project search option didn't get any , so just checked on the master branch as this plugin was missing and hence went on coding for new one and shared PR .

**Comment by poornimakshirsagar** [comment]
@bmr-cymru went through  points which you have mentioned 
>>Place the plugin in the storage profile :

isn't this an openstack plugin ? 
Manila is the "Shared File Systems as a Service" portion of OpenStack. Originally forked from Cinder, it is a way of presenting shared file systems to other services in OpenStack, such as a Nova Instance. 
See Manila http://docs.openstack.org/cli-reference/manila.html
  
>>
add_cmd_output("manilla list")
Check for OS_* environment variables (I assume these are used for manilla list?)
>>

command "manila list " just list NAS shares with filters. I have just checked checked the output:
<snip>
# manila list
+----+------+------+-------------+--------+-----------+-----------------+------+-------------------+
| ID | Name | Size | Share Proto | Status | Is Public | Share Type Name | Host | Availability Zone |
+----+------+------+-------------+--------+-----------+-----------------+------+-------------------+
+----+------+------+-------------+--------+-----------+-----------------+------+-------------------+
</snip>

is it really required ? 
in my opinion lets get precise manila data in sos.

**Comment by bmr-cymru** [comment]
> isn't this an openstack plugin ?

Yes: a plugin can be a member of multiple profiles - it is not an ownership relation.

> Manila is the "Shared File Systems as a Service" portion of OpenStack

Isn't a file system storage? ;-)

> is it really required ?
> in my opinion lets get precise manila data in sos.

Why isn't it required? What is "imprecise" about it?

I have no familiarity with Manila, but @ervikrant06 seemed to think it useful and I don't quite understand your argument against it - unless you mean that without authentication details it cannot collect useful data? That's unfortunate but a property of a number of plugins today - things that need authentication, need authentication.

**Comment by poornimakshirsagar** [comment]
>>
Yes: a plugin can be a member of multiple profiles - it is not an ownership relation.
Isn't a file system storage? ;-)

perfect:+1:  we can have this in both profiles openstack and storage .
>>
Why isn't it required? What is "imprecise" about it?
I have no familiarity with Manila, but @ervikrant06 seemed to think it useful and I don't quite understand your argument against it - unless you mean that without authentication details it cannot collect useful data?
>> 
Thanks @bmr-cymru  this was just my opinion ;-) 
If @ervikrant06  seemed to think its useful lets move ahead with it.

**Comment by bmr-cymru** [comment]
If it's costly then we might want to make it optional - as I said, I'm pretty ignorant of Manila, so I am relying on your and @ervikrant06 for expertise here :-)

**Comment by ervikrant06** [comment]
i am okay if we are ignoring it.

**Comment by psachin** [comment]
@poornimakshirsagar @ervikrant06 

Sorry to interrupt guys but I'm following this comments from the beginning and I think we should keep "list" command as sometimes it is necessary to ask this information from customer. It won't hurt as it is optional anyways.

**Comment by poornimakshirsagar** [comment]
cool @psachin let me dig a bit more into this and other manila related commands .

**Comment by poornimakshirsagar** [comment]
@psachin the list option don't seem to be mandatory.If we see other plugins "cinder list" is not captured as we have logs related to it already captured.

**Comment by poornimakshirsagar** [comment]
@bmr-cymru verified the output files, data and captured by this plugins with manila SME's gave go green for this plugin results rhos-manila.

**Comment by psachin** [comment]
@poornimakshirsagar 

If we have same info(in logs) as we get in `manila list`, it is great(I don't have env right now). Our concern is we don't end up asking output of `manila list` after asking sosreport. You see, I just want to avoid too-fro conversation.

**Comment by poornimakshirsagar** [comment]
agreed on too-fro conversations I have shared the output of manila list from my test box:

manila list

+----+------+------+-------------+--------+-----------+-----------------+------+-------------------+
| ID | Name | Size | Share Proto | Status | Is Public | Share Type Name | Host | Availability Zone |
+----+------+------+-------------+--------+-----------+-----------------+------+-------------------+
+----+------+------+-------------+--------+-----------+-----------------+------+-------------------+
hence investigated with SME if you know 'manila list' is a command that exercises a basic core api which will be called repeatedly when running the core tempest api tests.  We don't
need to separately invoke it I think 
if you still feel its required you can share a PR for the same 👍 .

**Comment by bmr-cymru** [comment]
Merged as commit 60895a4c5ee363e1c457a16e731fc56851b1ded2, thanks!

**Pull request closed without merging**